test/e2e/upgrade_management_cluster_test.go

```diff
@@ -152,3 +154,35 @@ func preUpgrade(clusterProxy framework.ClusterProxy) {
 	upgradeIronic(clusterProxy.GetClientSet())
 	upgradeBMO(clusterProxy.GetClientSet())
 }
+
+// preCleanupManagementCluster hook should be called from ClusterctlUpgradeSpec before cleaning the target management cluster
+// it moves back Ironic to the bootstrap cluster.
+func preCleanupManagementCluster(clusterProxy framework.ClusterProxy) {
+	// Remove ironic
+	removeIronic := func() {
+		By("Remove Ironic containers from the target cluster")
+		deploymentName := e2eConfig.GetVariable("NAMEPREFIX") + "-ironic"
+		ironicNamespace := e2eConfig.GetVariable("IRONIC_NAMESPACE")
+		err := clusterProxy.GetClientSet().AppsV1().Deployments(ironicNamespace).Delete(ctx, deploymentName, metav1.DeleteOptions{})
+		Expect(err).To(BeNil(), "Failed to delete Ironic from the target cluster")
+	}
```

Comment:
It would be nicer to re-use [removeIronicDeploymentOnTarget](https://github.com/metal3-io/cluster-api-provider-metal3/blob/main/test/e2e/pivoting_test.go#L217-L222) instead of creating a new function for this